File: 0507-perfect-number/0507-perfect-number.c
bool checkPerfectNumber(int num) {
    int total=0;
    for(int i=1;i<=num/2;i++){
        if(num%i==0){
            total+=i;
        }
    }
    if(total==num)
    return true;
    else
    return false;
}

File: 0507-perfect-number/0507-perfect-number.java
class Solution {
    public boolean checkPerfectNumber(int num) {
     int divisorsum=0;
     for(int itr=1;itr<=num/2;itr++){
        divisorsum += (num%itr==0)?itr:0;
     }
     return num==divisorsum?true:false;  
    }
}